package/MDAnalysis/analysis/bat.py

```diff
@@ -0,0 +1,422 @@
+
+
+class BAT(AnalysisBase):
+    """Calculate BAT coordinates for the specified AtomGroup.
+
+    BAT coordinates will be computed for the group of atoms and all frames
+    in the trajectory belonging to `ag'.`
+
+    """
+    def __init__(self, ag, initial_atom=None, **kwargs):
+        r"""Parameters
+        ----------
+        ag : AtomGroup or Universe
+            Group of atoms for which the BAT coordinates are calculated.
+            ag must have a bonds attribute. If not available,
+            bonds may be guessed using
+            :meth:`AtomGroup.guess_bonds <MDAnalysis.core.groups.AtomGroup.guess_bonds>`.
+            ag must only include one molecule.
+            If a trajectory is associated with the atoms then the computation
+            iterates over the trajectory.
+        initial_atom : Atom
+            The atom whose Cartesian coordinates define the translation
+            of the molecule. If not specified, the heaviest terminal atom
+            will be selected.
+
+        Raises
+        ------
+        AttributeError
+            If ag does not contain a bonds attribute
+        ValueError
+            If ag contains more than one molecule
+
+        """
+        super(BAT, self).__init__(ag.universe.trajectory, **kwargs)
+        self._ag = ag
+
+        # Check that the ag contains bonds
+        if not hasattr(ag, 'bonds'):
+            raise AttributeError('AtomGroup has no attribute bonds')
+
+        # Determine the root
+        # The initial atom must be a terminal atom
+        terminal_atoms = _sort_atoms_by_mass(\
+          [a for a in self._ag.atoms if len(a.bonds)==1])
+        if (initial_atom is None):
+            # Select the heaviest root atoms from the heaviest terminal atom
+            initial_atom = terminal_atoms[-1]
+        elif (not initial_atom in terminal_atoms):
+            raise Exception('Initial atom is not a terminal atom')
+        # The next atom in the root is bonded to the initial atom
+        second_atom = initial_atom.bonded_atoms[0]
+        # The last atom in the root is the heaviest atom
+        # bonded to the second atom
+        third_atom = [a for a in second_atom.bonded_atoms \
+          if (a in self._ag) and (a!=initial_atom)][-1]
+        self._root = mda.AtomGroup([initial_atom, second_atom, third_atom])
+
+        # Construct a list of torsion angles
+        self._torsions = []
+        selected_atoms = mda.AtomGroup(self._root)
+        while len(selected_atoms) < self._ag.n_atoms:
+            try:
+                torsion = _find_torsion(selected_atoms, self._ag)
+            except:
+                raise ValueError('AtomGroup is more than one molecule')
+
+            self._torsions.append(torsion)
+            selected_atoms += torsion[0]
+
+        # Get indices of the root and torsion atoms
+        # in a Cartesian positions array that matches the AtomGroup
+        ag_top_inds = list(self._ag.indices)
```

Comment:
`AtomGroup.indices` is going to be a numpy array, so it seems a little unwise to convert this to a list for the sake of the `.index()` method.  Instead you can use `np.where` and not have to make a copy of all the indices

Reply:
This is another trick I was unaware of. The np.where help recommends the use of np.nonzero, so I'm using that instead.